nerd-icons.el

```diff
@@ -1064,19 +1064,18 @@
 
 (defun nerd-icons-auto-mode-match? (&optional file)
   "Whether or not FILE's `major-mode' match against its `auto-mode-alist'."
-  (let* ((file (or file (buffer-file-name) (buffer-name)))
-         (auto-mode (nerd-icons-match-to-alist file auto-mode-alist)))
-    (eq major-mode auto-mode)))
+  (unless file (setq file (buffer-file-name)))
+  (when file (nerd-icons--auto-mode-match-internal file)))
 
-(defvar nerd-icons--file-cache (make-hash-table :test 'equal)
-  "Cache for file extension to mode mapping.")
+(defun nerd-icons--auto-mode-match-internal (file)
+  "Internal logic behind `nerd-icons-auto-mode-match?'.
+This function is split out so the result can be cached with `nerd-icons-cache'.
+FILE is the same as in `nerd-icons-auto-mode-match?', which see."
+  (eq major-mode (nerd-icons-match-to-alist file auto-mode-alist)))
```

Comment:
This needs some more discussion:

1. It checks the buffer's major-mode against the file-name, so accepting FILE as an argument doesn't make sense.
2. It's very brittle (incorrect).
    1. `auto-mode-alist` can contain arbitrary functions in addition to modes (i.e., functions that select a mode based on the file name).
    2. It doesn't handle major-mode remapping, so it won't work correctly with treesitter modes.

IMO, we should just remove `nerd-icons-auto-mode-match?` entirely. If a buffer has a file-name and that file has an icon, use that. Otherwise, use the major-mode's icon.

Thoughts?